From 42863d56e268ccfe22b7e064eddd8e9b9c4a2a57 Mon Sep 17 00:00:00 2001
From: effects3d <effects3dee@gmail.com>
Date: Thu, 27 Aug 2026 00:51:09 +0200
Subject: [PATCH 05/10] ntdll: Set MIMALLOC_FORCE_REDIRECT=1 by default

Signed-off-by: effects3d <effects3dee@gmail.com>
---
 dlls/ntdll/unix/loader.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/dlls/ntdll/unix/loader.c b/dlls/ntdll/unix/loader.c
index f67fa07d366..2fe89207609 100644
--- a/dlls/ntdll/unix/loader.c
+++ b/dlls/ntdll/unix/loader.c
@@ -2284,6 +2284,13 @@ BOOL alert_simulate_sched_quantum;
 BOOL fsync_simulate_sched_quantum;
 BOOL fsync_yield_to_waiters;
 
+static void mimalloc_force_redirect(void)
+{
+    if (!getenv("MIMALLOC_FORCE_REDIRECT")) {
+        setenv("MIMALLOC_FORCE_REDIRECT", "1", 0 );
+    }
+}
+
 static void hacks_init(void)
 {
     const char *sgi = getenv( "SteamGameId" );
@@ -2443,6 +2450,7 @@ static void start_main_thread(void)
     signal_init_threading();
     dbg_init();
     startup_info_size = server_init_process();
+    mimalloc_force_redirect();
     hacks_init();
     virtual_map_user_shared_data();
     init_cpu_info();
-- 
2.55.0

