From eb84197be2d6a03c80c793852e7a9f95e3a94e0b Mon Sep 17 00:00:00 2001
From: effects3d <effects3dee@gmail.com>
Date: Fri, 14 Aug 2026 21:57:58 +0200
Subject: [PATCH 04/14] wine.inf: Add Kombucha-specific registry overrides

Signed-off-by: effects3d <effects3dee@gmail.com>
---
 loader/wine.inf.in | 29 +++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index 738e98f4a83..c7c1aa93b0c 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -63,6 +63,7 @@ AddReg=\
     Debugger,\
     DirectX,\
     Fonts,\
+    KombuchaOverrides,\
     LicenseInformation,\
     MCI,\
     Misc,\
@@ -1343,3 +1344,31 @@ InfFiles  = 17
 NlsFiles  = 11
 SortFiles = 10,globalization\sorting
 WinmdFiles = 11,winmetadata
+
+[KombuchaOverrides]
+; Prefer loading native msvcrt DLLs that are bundled with Studio and WebView2 for better compatibility with Wine.
+; DLLs starting with 'api-ms-win-' cannot be overridden (?).
+; Wine always prefers loading builtin ucrtbase, so the native DLL would have to be copied over to the Wineprefix.
+HKCU,Software\Wine\DllOverrides,"api-ms-win-crt-conio-l1-1-0",0x2,"native,builtin"
+HKCU,Software\Wine\DllOverrides,"api-ms-win-crt-heap-l1-1-0",0x2,"native,builtin"
+HKCU,Software\Wine\DllOverrides,"api-ms-win-crt-locale-l1-1-0",0x2,"native,builtin"
+HKCU,Software\Wine\DllOverrides,"api-ms-win-crt-math-l1-1-0",0x2,"native,builtin"
+HKCU,Software\Wine\DllOverrides,"api-ms-win-crt-runtime-l1-1-0",0x2,"native,builtin"
+HKCU,Software\Wine\DllOverrides,"api-ms-win-crt-stdio-l1-1-0",0x2,"native,builtin"
+HKCU,Software\Wine\DllOverrides,"api-ms-win-crt-time-l1-1-0",0x2,"native,builtin"
+HKCU,Software\Wine\DllOverrides,"concrt140",0x2,"native,builtin"
+HKCU,Software\Wine\DllOverrides,"msvcp140",0x2,"native,builtin"
+HKCU,Software\Wine\DllOverrides,"msvcp140_1",0x2,"native,builtin"
+HKCU,Software\Wine\DllOverrides,"msvcp140_2",0x2,"native,builtin"
+HKCU,Software\Wine\DllOverrides,"msvcp140_codecvt_ids",0x2,"native,builtin"
+HKCU,Software\Wine\DllOverrides,"ucrtbase",0x2,"native,builtin"
+HKCU,Software\Wine\DllOverrides,"vccorlib140",0x2,"native,builtin"
+HKCU,Software\Wine\DllOverrides,"vcruntime140",0x2,"native,builtin"
+HKCU,Software\Wine\DllOverrides,"vcruntime140_1",0x2,"native,builtin"
+; Force HiDPI awareness to fix broken splash screen and docking overlay on DPI values >96.
+; Probably fixed with Wine commit a37867ddf8418439b37e062abe156a42accf4d32 but there shouldn't be any harm in keeping it.
+HKCU,Software\Microsoft\Windows NT\CurrentVersion\AppCompatFlags\Layers,"",0x2,"~ HIGHDPIAWARE"
+; WebView2 requires DirectComposition on Windows versions >=8.1, which Wine doesn't support as of 2026-08-11, so spoof the Windows version to 8.
+HKCU,Software\Wine\AppDefaults\msedgewebview2.exe,"Version",0x2,"win8"
+; WebView2 uses Arial as its font, which isn't available by default on most Linux distributions, so force it to use Liberation Sans.
+HKCU,Software\Wine\Fonts\Replacements,"Arial",0x2,"Liberation Sans"
-- 
2.55.0

