From dcf36845c0bcc2951ba6274061017292b0bb25f4 Mon Sep 17 00:00:00 2001
From: effects3d <effects3dee@gmail.com>
Date: Sun, 30 Aug 2026 22:11:15 +0200
Subject: [PATCH 07/18] server: Notify if ntsync is unavailable

Signed-off-by: effects3d <effects3dee@gmail.com>
---
 server/inproc_sync.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/server/inproc_sync.c b/server/inproc_sync.c
index 522d9a4..05d240a 100644
--- a/server/inproc_sync.c
+++ b/server/inproc_sync.c
@@ -47,7 +47,13 @@
 int get_inproc_device_fd(void)
 {
     static int fd = -2;
-    if (fd == -2) fd = open( "/dev/ntsync", O_CLOEXEC | O_RDONLY );
+    if (fd == -2) {
+        fd = open( "/dev/ntsync", O_CLOEXEC | O_RDONLY );
+
+        if ( fd < 0 ) {
+            fprintf( stderr, "kombucha: ntsync is unavailable, performance will be affected.\n");
+        }
+    }
     return fd;
 }
 
-- 
2.56.0

