From a912fb08058b5ef6413d3e8513a19d06190cee44 Mon Sep 17 00:00:00 2001
From: effects3d <effects3dee@gmail.com>
Date: Mon, 17 Aug 2026 15:06:54 +0200
Subject: [PATCH 13/13] dlls/ntdll: Force mimalloc redirection

Signed-off-by: effects3d <effects3dee@gmail.com>
---
 dlls/ntdll/unix/loader.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/dlls/ntdll/unix/loader.c b/dlls/ntdll/unix/loader.c
index 0d43c5c..3ff2e14 100644
--- a/dlls/ntdll/unix/loader.c
+++ b/dlls/ntdll/unix/loader.c
@@ -1855,6 +1855,11 @@ static ULONG_PTR get_image_address(void)
     return 0;
 }
 
+static void mimalloc_force_redirect(void)
+{
+    setenv( "MIMALLOC_FORCE_REDIRECT", "1", 0 );
+}
+
 /***********************************************************************
  *           start_main_thread
  */
@@ -1864,6 +1869,7 @@ static void start_main_thread(void)
 
     dbg_init();
     startup_info_size = server_init_process();
+    mimalloc_force_redirect();
     virtual_map_user_shared_data();
     init_cpu_info();
     init_files();
-- 
2.55.0

